middle_end/flambda2/simplify/simplify_extcall.ml

```diff
@@ -129,7 +130,41 @@ let simplify_comparison ~dbg ~dacc ~cont ~tagged_prim ~float_prim
         ( Untagged_immediate | Naked_float | Naked_nativeint | Naked_int32
         | Naked_int64 ),
       (Unknown | Invalid | Wrong_kind) ) ->
-    Unchanged
+    Unchanged { return_types = None }
+
+let simplify_caml_make_vect dacc ~len_ty ~init_value_ty : t =
+  let typing_env = DA.typing_env dacc in
+  let element_kind : _ Or_unknown.t Or_bottom.t =
+    (* We can't deduce subkind information, e.g. an array is all-immediates
+       rather than arbitrary values, but we can deduce kind information. *)
+    if not (Flambda_features.flat_float_array ())
+    then
+      Ok
+        (Known
+           (Flambda_kind.With_subkind.create (T.kind init_value_ty) Anything))
+    else
+      match T.prove_is_or_is_not_a_boxed_float typing_env init_value_ty with
+      | Proved true ->
+        (* A boxed float provided to [caml_make_vect] with the float array
+           optimisation on will always yield a flat array of naked floats. *)
+        Ok (Known Flambda_kind.With_subkind.naked_float)
+      | Proved false | Unknown -> Ok Unknown
```

Comment:
Since that would require inferring subkind information, which we cannot do -- we can only infer the kind information.  (Consider for example an array of `option` values, where some are immediates and some are not.)